util/menu_iterator.lua

```diff
@@ -62,7 +63,72 @@ local function iterate(menu, timeout, icon)
     }).id
 end
 
-return { iterate = iterate }
+-- Generates a menu compatible with the iterate function argument and suitable
+-- for the following cases: * all possible choices are the hosts themselves
+-- * all possible choices are all the possible subsets of the set of hosts (the
+--   powerset)
+--
+-- The following describes the function arguments:
+-- * args: an array containing the following members:
+--   * hosts:         the list of hosts from which to generate the menu
+--   * name:          the displayed name of the menu (in the form "name: choices")
+--   * selected_cb:   the callback to execute for each selected host. Takes the
+--                    host as a string argument.
+--   * rejected_cb:   the callback to execute for each rejected host (in the set
+--                    of possible hosts, but not selected). Takes the host as a
+--                    string argument.
+--   * extra_choices: an array of pairs { choice_text, cb } for extra choices to
+--                    be added to the menu.
+--   * combination:   the combination of host to generate. Possible choices are
+--                    "powerset" and "single" (the default).
+local function menu(args)
+    local hosts       = assert(args.hosts or args[1])
+    local name        = assert(args.name or args[2])
+    local selected_cb = args.selected_cb
```

Comment:
What if `args` doesn't have `selected_cb`? Put a fallback, for instance:

```lua
local selected_cb = args.selected_cb or function() end
```

Reply:
This is actually checked [here](https://github.com/lcpz/lain/blob/9e21a95a1a20a929a4a0f62f3ffddeeb6e041045/util/menu_iterator.lua#L22). Empty callbacks are used, for instance like [this][example].

[example]: https://github.com/sim590/amh/commit/c648aa927649e8c5388362e4104afe04be672d48#diff-42b565af082eb6b5eca071ecb51c0225R36

Reply:
This indicates that there are things you didn't explain. You have to do it, either by adding comments or writing a proper wiki entry, in which you should also put some usage examples (see [here](https://github.com/lcpz/lain/pull/377#issuecomment-366209236)).